lib/puppet/settings/base_setting.rb

```diff
@@ -151,7 +161,11 @@ def set_meta(meta)
   end
 
   def deprecated=(deprecation)
-    raise(ArgumentError, "'#{deprecation}' is an unknown setting deprecation state.  Must be either :completely or :allowed_on_commandline") unless [:completely, :allowed_on_commandline].include?(deprecation)
+    unless [:completely, :allowed_on_commandline].include?(deprecation)
+      # 'deprecated' is a Puppet setting and ':completely' and ':allowed_on_commandline' are possible values and should not be translated
+      raise ArgumentError, _("Unsupported deprecated value '%{deprecation}'.") % { deprecation: deprecation } +
```

Comment:
I think "Unsupported deprecated value" is supposed to be "Unsupported deprecation state" - these values describe the way we should handle another setting that has been deprecated in some way.

Reply:
Actually the thing you set is "deprecated". Here is an example from defaults.rb

```
  define_settings(:main,
      # Whether the application management feature is on or off - now deprecated and always on.
      :app_management => {
          :default  => false,
          :type     => :boolean,
          :desc       => "This setting has no effect and will be removed in a future Puppet version.",
          :deprecated => :completely,
      }
  )
```
Which is why 'deprecated' is called out as a Puppet setting in the TRANSLATOR comment

Reply:
OK, I understand the change you made, and I agree that it is a clearer message.  👍